test/e2e/metrics_kyma_input_test.go

```diff
@@ -0,0 +1,172 @@
+
+		backendForNoKymaInput := backend.New(mockNs, backend.SignalTypeMetrics, backend.WithName(backendForNoKymaInputName))
+		objs = append(objs, backendForNoKymaInput.K8sObjects()...)
+		backendForNoKymaInputExportURL = backendForNoKymaInput.ExportURL(proxyClient)
+
+		metricPipelineWithAnnotation := testutils.NewMetricPipelineBuilder().
+			WithName(pipelineWithAnnotationName).
+			WithAnnotations(map[string]string{gateway.KymaInputAnnotation: "true"}).
+			WithOTLPOutput(testutils.OTLPEndpoint(backendForKymaInput.Endpoint())).
+			Build()
+		objs = append(objs, &metricPipelineWithAnnotation)
+
+		metricPipelineWithoutAnnotation := testutils.NewMetricPipelineBuilder().
+			WithName(pipelineWithoutAnnotationName).
+			WithOTLPOutput(testutils.OTLPEndpoint(backendForNoKymaInput.Endpoint())).
+			Build()
+		objs = append(objs, &metricPipelineWithoutAnnotation)
+
+		return objs
+	}
+
+	BeforeAll(func() {
+		k8sObjects := makeResources()
+
+		DeferCleanup(func() {
+			Expect(kitk8s.DeleteObjects(ctx, k8sClient, k8sObjects...)).Should(Succeed())
+		})
+
+		Expect(kitk8s.CreateObjects(ctx, k8sClient, k8sObjects...)).Should(Succeed())
+	})
+
+	Context("When a metricpipeline with kyma input annotation exists", Ordered, func() {
+
+		It("Ensures the metric gateway deployment is ready", func() {
+			assert.DeploymentReady(ctx, k8sClient, kitkyma.MetricGatewayName)
+		})
+
+		It("Ensures the metrics backends are ready", func() {
+			assert.DeploymentReady(ctx, k8sClient, types.NamespacedName{Name: backendForKymaInputName, Namespace: mockNs})
+			assert.DeploymentReady(ctx, k8sClient, types.NamespacedName{Name: backendForNoKymaInputName, Namespace: mockNs})
+		})
+
+		It("Ensures the metric pipelines are healthy", func() {
+			assert.MetricPipelineHealthy(ctx, k8sClient, pipelineWithAnnotationName)
+			assert.MetricPipelineHealthy(ctx, k8sClient, pipelineWithoutAnnotationName)
+		})
+
+		It("Ensures Telemetry module status metrics are sent to the backend which is receiving metrics from the pipeline with annotation", func() {
+			Eventually(func(g Gomega) {
+				resp, err := proxyClient.Get(backendForKymaInputExportURL)
+				g.Expect(err).NotTo(HaveOccurred())
+				g.Expect(resp).To(HaveHTTPStatus(http.StatusOK))
+				g.Expect(resp).To(HaveHTTPBody(
+					WithFlatMetrics(SatisfyAll(
+						ContainElement(SatisfyAll(
+							// Check the "kyma.module.status.state" metric
+							HaveField("Name", "kyma.module.status.state"),
+							HaveField("MetricAttributes", HaveKey("state")),
+							HaveField("ResourceAttributes", HaveKeyWithValue("k8s.namespace.name", kitkyma.SystemNamespaceName)),
+							HaveField("ResourceAttributes", HaveKeyWithValue("kyma.module.name", "Telemetry")),
+							HaveField("ScopeAndVersion", HaveField("Name", metric.InstrumentationScopeKyma)),
+							HaveField("ScopeAndVersion", HaveField("Version", SatisfyAny(
+								Equal("main"),
+								MatchRegexp("[0-9]+.[0-9]+.[0-9]+"),
+							))),
+						)),
```

Comment:
Can we maybe split this into 4 consequent checks, each check including `ContainElement` to avoid deep nesting

Reply:
Done